### DISCUSSION TITLE
what is similar to requests in python sdk to C++ sdk

### DISCUSSION THREAD

**Discussion by tagopinath** [aws/aws-sdk-cpp#2797] [Q&A] [answered]
Hello,

I am new to AWS sdk. we have a plugin written in C++. 
Now, we need to integrate Aws for some customer to get / put some requests in one of the aws cloud storage.
for that, i am looking into AWS sdk c++.

we have a sample in python, need to write same like in c++ (both - aws sdk)

in python, i came across requests module to post and get some details with url and some parameters.

i am looking for the same module in c++ sdk.

please suggest if anyone came across.

Regards,
Gopi.

**Comment by sbiscigl** [comment, marked as answer]
Ah i take this is related to [issues/2802](https://github.com/aws/aws-sdk-cpp/issues/2802). there is two questions here

> what is similar to requests in python sdk to C++

C++ like python has http clients, if you are looking for one that is like python requests i would recommend [curl for people](https://github.com/libcpr/cpr) which in its description it says

`C++ Requests is a simple wrapper around [libcurl](http://curl.haxx.se/libcurl) inspired by the excellent [Python Requests](https://github.com/kennethreitz/requests) project.`

> Now, we need to integrate Aws for some customer to get / put some requests in one of the aws cloud storage.

which cloud storage? S3? if so you can see some examples on how use S3 in the doc examples project. 

* [S3 get object example](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/cpp/example_code/s3/get_object.cpp)
* [S3 put object example](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/cpp/example_code/s3/put_object.cpp)

_IF_ you plan on calling S3 you should use the SDK client and _NOT_ curl for people
